plugins/modules/jenkins_node.py

```diff
@@ -0,0 +1,393 @@
+                    raise e
+
+                # TODO: Remove authorization workaround.
+                self.result['warnings'].append(
+                    "suppressed 401 Not Authorized on redirect after node deleted: see https://review.opendev.org/c/jjb/python-jenkins/+/931707"
+                )
+
+        present = self.instance.node_exists(self.name)
+        deleted = False
+        if present:
+            if not self.module.check_mode:
+                delete_node()
+
+            deleted = True
+
+        self.result['deleted'] = deleted
+        if deleted:
+            self.result['changed'] = True
+
+    def enabled_node(self):
+        present = self.present_node()
+
+        enabled = False
+
+        if present:
+            info = self.instance.get_node_info(self.name)
+
+            if info['offline']:
+                if not self.module.check_mode:
+                    self.instance.enable_node(self.name)
+
+                enabled = True
+        else:
+            # Would have created node with initial state enabled therefore would not have
+            # needed to enable therefore not enabled.
+            if not self.module.check_mode:
+                raise Exception("enabled_node present is False outside of check mode")
+            enabled = False
+
+        self.result['enabled'] = enabled
+        if enabled:
+            self.result['changed'] = True
+
+    def disabled_node(self):
+        present = self.present_node()
+
+        disabled = False
+
+        if present:
+            info = self.instance.get_node_info(self.name)
+
+            if not info['offline']:
+                if not self.module.check_mode:
+                    self.instance.disable_node(self.name)
+
+                disabled = True
+        else:
+            # Would have created node with initial state enabled therefore would have
+            # needed to disable therefore disabled.
+            if not self.module.check_mode:
+                raise Exception("disabled_node present is False outside of check mode")
+            disabled = True
+
+        self.result['disabled'] = disabled
+        if disabled:
+            self.result['changed'] = True
+
+
+def test_dependencies(module):
+    if not python_jenkins_installed:
+        module.fail_json(
+            msg=missing_required_lib("python-jenkins", url="https://python-jenkins.readthedocs.io/en/latest/install.html"),
+            exception=JENKINS_IMP_ERR
+        )
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True, type='str'),
+            url=dict(default='http://localhost:8080'),
+            user=dict(),
+            token=dict(no_log=True),
+            state=dict(choices=['enabled', 'disabled', 'present', 'absent'], default='present'),
+            num_executors=dict(type='int'),
+            labels=dict(type='list', elements='str'),
+        ),
+        supports_check_mode=True,
+    )
+
+    test_dependencies(module)
```

Comment:
You might want to consider using `module_utils.deps`, see https://docs.ansible.com/ansible/latest/collections/community/general/docsite/guide_deps.html for guidance on how to use it.